### PR TITLE
Add ListExternalGroupsForTeamBySlug to Teams API

### PR DESCRIPTION
**Pull request by bodgit**
#2217 added support for the majority of the External Groups API but omitted listing groups for a specific Team within an Organization.

This PR adds the missing method, and corrects the comment on the original `ListExternalGroups` method as that actually returns all external groups within the Organization. I added the `ForTeamBySlug` suffix based on other methods in the source to distinguish it from the original method.

I've added tests based on the original `ListExternalGroups` method and everything passes.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @bodgit !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by bodgit** [comment]
@gmlewis good to merge now?

**Pull request merged**
